Makefile

```
@@ -129,3 +135,9 @@ image-update:
 soc:
 	@echo Version: $(VERSION) $(SHA) $(BUILD_DATE)
 	go build $(BUILD_TAGS) $(BUILD_ARGS) github.com/evcc-io/evcc/cmd/soc
+
+# patch asn1.go to allow Elli buggy certificates to be accepted with EEBUS
+patch-asn1:
+ifeq ($(shell test -e $(ASN1_FILE) && echo 1),1)
+	sudo patch -N -t -d $(ASN1_PATH) -i $(CURRENT_DIR)/patch/asn1.diff
```

Move sudo to calling location?

---

Dockerfile fehlt noch?

---

Können wir den Patch-PR vllt separieren und separat testen?

---

Na ja, dann hast mit dem PR hier die templates von Elli drin und werden in configure angezeigt, aber die nightly oder release builds unterstützen das nicht.

Ich würde es daher präferieren das hier alles zusammen abzuhandeln

---

What exactly do you mean by this: `Move sudo to calling location?` ?

---

Ich würde vermuten: Das `sudo` hier an dieser Stelle weg und das ganze Makefile ggf. mit sudo (oder anderen ausreichenden Rechten) ausführen.
```suggestion
	patch -N -t -d $(ASN1_PATH) -i $(CURRENT_DIR)/patch/asn1.diff
```

---

Hmm .... In Zeile 125 wird es ebenfalls schon genau so verwendet. Den make bin sudo aufrufen ... fände ich jetzt merkwürdig

---

Da in docker build normal kein `sudo` vorhanden ist, würde der Aufruf in `Dockerfile` nicht funktionieren.

---

Ok, also Aufruf mit `sudo` in den CI workflows only

---

Ich habe das mal hier geändert: https://github.com/evcc-io/evcc/pull/4207/commits/e7a609b74782e3d832ae334494ba08ed1303e467

Und das patchen noch zum `Build` job hinzugefügt, aber ich vermute mal das ist unnötig weil der `clean` job ja immer davor ausgeführt werden sollte